vault/resource_secrets_sync_gh_apps_destination.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	syncutil "github.com/hashicorp/terraform-provider-vault/internal/sync"
+)
+
+const (
+	ghAppsSyncType = "github-apps"
+)
+
+var githubAppsSyncWriteFields = []string{
+	consts.FieldName,
+	consts.FieldAppID,
+	consts.FieldPrivateKey,
+}
+
+var githubAppsSyncReadFields = []string{
```

Comment:
A `fingerprint` of the private key is also returned

Reply:
Still working on this, for some reason I get a panic when I try to read the fingerprint.

Reply:
Sounds good! A fingerprint is only returned when a valid key is provided.